### ISSUE TITLE
Segfault  with smbios-keyboard-ctl

### ISSUE THREAD

**Issue by mshindal** [dell/libsmbios#1]
Hey, so I'm trying to use libsmbios on my Dell XPS 13 Project Sputnik machine. When I run `smbios-keyboard-ctl -g` or `smbios-keyboard-ctl -i` I get a seg fault. I'd like to give you a stack trace but I'm not sure how to build libsmbios with debugging symbols.

**Comment by superm1** [comment]
1. Which XPS 13 model?

1. BIOS version?

1. libsmbios version?

1. Packaged from Ubuntu repositories or hand installed?

1. Could you provide output from when you run those commands?

Fortunately smbios-keyboard-ctl is a python application, so hopefully running like this should get more verbose output from before the crash:
`# sudo smbios-keyboard-ctl -i -v`

**Comment by mshindal** [comment]
1. XPS 9333 Developer Edition
2. A02
3. I checked out from master, so 2.3.1
4. Hand installed
5. 
```
[matt@localhost libsmbios]$ smbios-keyboard-ctl -g
Helper function to print current status of keyboard illumination

 Current status of KeyBoard Illumination setting on your system: 
-------------------------------------------------------------------
Segmentation fault (core dumped)
```
```
[matt@localhost libsmbios]$ smbios-keyboard-ctl -i -v
Libsmbios version : 2.3.1
smbios-keyboard-ctl version : 2.3.1

 Capabilities of KeyBoard Illumination on your system: 
-------------------------------------------------------------------
Segmentation fault (core dumped)
```

**Comment by mshindal** [comment]
After playing around with smbios-keyboard-ctl some more, I noticed some weird things.
* When I try to run it with sudo, I get a "command not found" error. 
* When I run it as a normal user, I get the segfault, as you can see in my previous comment
* When I switch to root, cd to the directory it installed to, and then run it from there, it works and does not segfault.

Example:
```
[matt@localhost libsmbios]$ su -
Password: 
[root@localhost ~]# cd /usr/local/sbin/
[root@localhost sbin]# ./smbios-keyboard-ctl -g
Helper function to print current status of keyboard illumination

 Current status of KeyBoard Illumination setting on your system: 
-------------------------------------------------------------------

 Configured mode state: 
	 Auto: Input-activity-based On; input-activity based Off

 Your Keyboard will illumination on: 
	Any Keystroke
	Touchpad activity

 Keyboard illumination timeout has bee set at: 0  Seconds

 Current setting of ALS value that turns the light on or off: 0
 Current ALS Reading : 0
 Current keyboard light level : 2
```

**Comment by superm1** [comment]
Those do sound like some pretty weird thing indeed.  What's your distro?  I think @prabhakarpujeri will need to reproduce all this to dig in a bit better.  I'm suspecting that some of the autotools configuration isn't getting done right.  Hasn't been reported yet since most people use libsmbios packages (debs or rpms).

There's a few problems I see going on here I think.
1. There is supposed to be a warning when running as a regular user that dell_rbu or dcdbas aren't loaded or can't be accessed.  Is it possible your disto's policies (apparmor, SELinux, etc) are causing a failure there?  Or were you running in a container that didn't have bind mounts or something?
2. The command not found makes me wonder if the python stuff wasn't set up in your path properly.

**Comment by mshindal** [comment]
I'm using Fedora 24. 

Fedora does indeed have SELinux enabled by default. Admittedly I don't really know much about SELinux but after some Googling it looks like the log should be in `/var/log/audit/audit.log`. The only entry in that log that looked related to `smbios-keyboard-ctl` was this one, which as far as I can tell simply says that the process ended abnormally:
```
type=ANOM_ABEND msg=audit(1479163485.971:484): auid=1000 uid=1000 gid=1000 ses=3 subj=unconfined_u:unconfined_r:unconfined_t:s0-s0:c0.c1023 pid=21972 comm="smbios-keyboard" exe="/usr/bin/python2.7" sig=11
```

**Comment by mshindal** [comment]
I should mention that all I want to do is make my keyboard backlight stay on. My laptop makes an annoying "coil whine" sound that only goes away when the keyboard backlight is on. Switching to the root user and then running `smbios-keyboard-ctl --set-mode` gets me close: I can set the keyboard state to OFF but setting it to ON or AUTO does nothing. The physical keyboard backlight button on my keyboard seems to toggle it between AUTO and OFF. So I have no way right now to set it to ON.

**Comment by daseinhorn** [comment]
I experience the same issue. But unlike @mshindal I'm not event able to set the state to OFF.
This is the output:
`root@workstation:/home/max smbios-keyboard-ctl --set-mode

Setting USER Selectable Keyboard Mode
Segmentation fault (core dumped)
`

This is really annoying. It does not even work with the shipped ubuntu.

**Comment by superm1** [comment]
Can you please try this with the branch referred to in https://github.com/dell/libsmbios/issues/3?  I believe it might actually be the same fix.

**Comment by superm1** [comment]
Closing at his should be the same fix as #3.  Please open a new issue if you're still affected.
